crates/bins/src/bin/datadog-static-analyzer.rs

```diff
@@ -767,36 +767,72 @@ fn main() -> Result<()> {
-                    .expect("path contains non-Unicode characters");
-                let res = if let Ok(file_content) = fs::read_to_string(path) {
-                    let file_content = Arc::from(file_content);
-                    find_secrets(
-                        &sds_scanner,
-                        &secrets_rules,
-                        relative_path,
-                        &file_content,
-                        &analysis_options,
-                    )
-                } else {
-                    // this is generally because the file is binary.
-                    if use_debug {
-                        eprintln!("error when getting content of path {}", &path.display());
+        (secrets_results, path_metadata) = secrets_files
+            .into_par_iter()
+            .fold(
+                || (Vec::new(), HashMap::new()),
+                |(_fold_results, mut path_metadata), path| {
+                    let relative_path = path
+                        .strip_prefix(directory_path)
+                        .unwrap()
+                        .to_str()
+                        .expect("path contains non-Unicode characters");
+                    let res = if let Ok(file_content) = fs::read_to_string(&path) {
+                        let file_content = Arc::from(file_content);
+                        let secrets = find_secrets(
+                            &sds_scanner,
+                            &secrets_rules,
+                            relative_path,
+                            &file_content,
+                            &analysis_options,
+                        );
+
+                        if !secrets.is_empty()
+                            && !all_path_metadata_secrets.contains_key(relative_path)
+                        {
+                            let cloned_path_str = relative_path.to_string();
+                            let language_opt = get_language_for_file(&path);
+
+                            if let Some(language) = language_opt {
+                                let metadata = if is_test_file(
+                                    language,
+                                    file_content.as_ref(),
+                                    std::path::Path::new(&cloned_path_str),
+                                    None,
+                                ) {
+                                    Some(ArtifactClassification { is_test_file: true })
+                                } else {
+                                    None
+                                };
+                                path_metadata.insert(cloned_path_str.clone(), metadata);
+                            }
+                        }
+
+                        secrets
+                    } else {
+                        // this is generally because the file is binary.
+                        if use_debug {
+                            eprintln!("error when getting content of path {}", &path.display());
```

Comment:
don't think you need to take a reference here:

```suggestion
                            eprintln!("error when getting content of path {}", path.display());
```